### DISCUSSION TITLE
A new (empty) git repository won't initialize properly

### DISCUSSION THREAD

**Discussion by hashworks** [requarks/wiki#3572] [Error / Bug Report]
**Describe the bug**
When a completely new git repository is added as storage it won't be initialized properly, resulting in a sync error.

**To Reproduce**
1. Create a new wiki
2. Create a new git without content / branches
3. Add git as storage to wiki
4. Create files and try to sync them

**Expected behavior**
The newly created branch is pushed and created in origin.

**Host Info (please complete the following information):**
 - OS: ArchLinux
 - Wiki.js version: 2.0.0-beta.303
 - Database engine: sqlite3

**Additional context**
This happens since [Wiki.js is using](https://github.com/Requarks/wiki/blob/cffd32dee0c8fd30a3f0c5e676146f6ba2ee392e/server/modules/storage/git/storage.js#L130) `git push` instead of `git push -u origin master`.

To resolve that one must sync manually once:
```
 % git push
No refs in common and none specified; doing nothing.
Perhaps you should specify a branch such as 'master'.
fatal: the remote end hung up unexpectedly
error
 % git push -u origin master
[..., works]
```

**Comment by hashworks** [comment]
Additionally: `simple-git-promise` is marked as [deprecated](https://github.com/renolc/simple-git-promise#readme).

**Comment by NGPixel** [comment]
We're not using `simple-git-promise` anywhere in the project...

**Comment by hashworks** [comment]
https://github.com/Requarks/wiki/blob/cffd32dee0c8fd30a3f0c5e676146f6ba2ee392e/server/modules/storage/git/storage.js#L2

**Comment by NGPixel** [comment]
Re-read the line you just posted... It's not `simple-git-promise`.

**Comment by hashworks** [comment]
Ah, I see. My bad!

**Comment by arturaum** [comment]
So, how to fix this? I need to `git init` somewhere or what?

**Comment by arturaum** [comment]
to fix this, you need to stop wiki.js service, then go to `./data/repo` folder then:
```
git init
git add -A
git commit -m "first commit"
git remote add origin git@github.com:username/repo-name.git
git push -u origin master
```
And thanks to God, sync finally starts working.

**Comment by nlfiedler** [comment]
Just ran into this issue, although the error in the log is now `warn: fatal: couldn't find remote ref master`. Internally, maybe you need to pass the `-u` flag to `git-push` on the first push? If I do this myself within the container then the sync seems to succeed (see also issue #2560).
